===================================================================
RCS file: /cvsroot/gaim/gaim/src/protocols/msn/object.c,v
retrieving revision 1.3
retrieving revision 1.4
diff -u -r1.3 -r1.4
--- gaim/gaim/src/protocols/msn/object.c	2004/08/19 01:13:58	1.3
+++ gaim/gaim/src/protocols/msn/object.c	2004/08/22 17:31:37	1.4
@@ -35,11 +35,18 @@
 	if ((tag = strstr(str, id "=\"")) != NULL) \
 	{ \
 		char buf[16]; \
+		size_t offset; \
 		tag += strlen(id "=\""); \
 		c = strchr(tag, '"'); \
-		strncpy(buf, tag, c - tag); \
-		buf[c - tag] = '\0'; \
-		obj->field = atoi(buf); \
+		if (c != NULL) \
+		{ \
+			memset(buf, 0, sizeof(buf)); \
+			offset = c - tag; \
+			if (offset >= sizeof(buf)) \
+				offset = sizeof(buf) - 1; \
+			strncpy(buf, tag, offset); \
+			obj->field = atoi(buf); \
+		} \
 	}
 
 static GList *local_objs;
